src/wazuh_modules/vulnerability_detector/wm_vuln_detector.c

```diff
@@ -1361,34 +1360,7 @@ int wm_vuldet_process_agent_vulnerabilities(sqlite3 *db, OSHash *cve_table, scan
                 continue;
             }
 
-            //Save the vulnerability in the agent database
-            bool check_pkg_existence = !strcmp(pkg->type, VULN_CVES_TYPE_PACKAGE);
-            cJSON* j_result = wdb_insert_vuln_cves(scan_ctx->agent_id, pkg->bin_name, pkg->version, pkg->arch, cve,
-                                                            pkg->reference, pkg->type, VULN_CVES_STATUS_VALID, check_pkg_existence, &sock);
-
-            bool success = FALSE;
-            bool update = FALSE;
-
-            if (j_result) {
-                cJSON* j_status = cJSON_GetObjectItem(j_result, "status");
-                success = (cJSON_IsString(j_status) && 0 == strcmp(j_status->valuestring, "SUCCESS"));
-                cJSON* j_action = cJSON_GetObjectItem(j_result, "action");
-                update = (cJSON_IsString(j_action) && 0 == strcmp(j_action->valuestring, "UPDATE"));
-                cJSON_Delete(j_result);
-            }
-            if (!success) {
-                mtdebug1(WM_VULNDETECTOR_LOGTAG, "Failed to insert %s for package %s in the agent %.3d database",
-                        cve ? cve : "null",
-                        pkg->reference ? pkg->reference : "null",
-                        scan_ctx->agent_id);
-            }
-
-            if (scan_ctx->scan_type == VU_BASELINE_SCAN || update) {
-                // No report should be generated for baseline scans or already reported vulnerabilities
-                pkg = next;
-                continue;
-            }
-
+            wm_vuldet_free_report(report);
```

Comment:
If you move the call of `wm_vuldet_free_report` to this place, you will cause always a memory leak of the last report being evaluated in the `do while` loop.

Reply:
Totally! Fixed now